app/controllers/manager/administrateurs_controller.rb

```diff
@@ -33,6 +33,21 @@ def enable_feature
       head :ok
     end
 
+    def delete
+      administrateur = Administrateur.find(params[:id])
+
+      if !administrateur.can_be_deleted?
+        fail "Cannot delete this administrateur because it has dossiers or procedures"
```

Comment:
J'ai l'impresion que ça va souvent arriver que la suppression d'un admin échoue (par exemple parce qu'il a déjà une démarche). J'ai l'impression qu'on gagnera du débug à présenter le message d'erreur au superadmin, non ?

```suggestion
        flash[:notice] = "L'administrateur #{administrateur.id} n’a pas pu être supprimé, car il possède des dossiers ou des démarches."
        return redirect_to manager_administrateur_path(administrateur)
```

(Faut vérifier le code, c'est du yolo 😅 )

Reply:
ça ne devrait pas arriver car le bouton est masqué si l'administrateur ne peut être supprimé (cf le template). Par contre je peux mettre un flash aussi !

Reply:
Oh, d'accord, je n'avais pas vu 😅 

Mmm, généralement j'ai l'impression que si une fonctionnalité n'est pas activée, il vaut mieux laisser le bouton et afficher une erreur (ou le désactiver avec un message). Si on le fait juste disparaître, l'utilisateur le cherche, et ne sait pas pourquoi il est plus là.

Du coup j'aurais une préférence pour "On laisse le bouton tout le temps, et on affiche une erreur claire au besoin". Mais comme tu le sens :)

Reply:
le bouton est toujours affiché, mais désactivé avec un tooltip lorsqu'il ne peut-être supprimé